packages/server/modules/workspaces/repositories/users.ts

```diff
@@ -12,3 +20,87 @@ export const setUserActiveWorkspaceFactory =
       meta.set(userId, 'isProjectsActive', isProjectsActive)
     ])
   }
+
+export const getInvitableCollaboratorsByProjectIdFactory =
+  ({ db }: { db: Knex }) =>
+  async ({
+    filter,
+    cursor,
+    limit
+  }: {
+    filter: {
+      workspaceId: string
+      projectId: string
+      search?: string
+    }
+    cursor?: string
+    limit: number
+  }): Promise<WorkspaceTeamMember[]> => {
```

Comment:
count & get items have the same base query that should be extracted out to its own function

Reply:
unfortunately I couldn't make it work since I need an async function to build the query because of the whereNotIn clause.
I belive the way knex works doesn't fit with async functions since the query is immediately resolved and I can't use the query builder.

Reply:
that async clause doesn't need to be async either